lib/patch-by-id.js

```diff
@@ -0,0 +1,187 @@
+// Copyright IBM Corp. 2015,2016. All Rights Reserved.
+// Node module: loopback-datasource-juggler
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+var ValidationError = require('./validations').ValidationError;
+var assert = require('assert');
+var dao = require('./dao');
+var utils = require('./utils');
+var idEquals = utils.idEquals;
+var removeUndefined = utils.removeUndefined;
+
+module.exports = function patchById(id, data, options, cb) {
+  var connectionPromise = dao.stillConnecting(this.getDataSource(), this, arguments);
+  if (connectionPromise) {
+    return connectionPromise;
+  }
+  assert(arguments.length >= 2, 'At least two arguments are required');
+
+  var Model = this;
+  var where = {};
+  var strict = this.settings.strict;
+  // TODO: the methods which we have imported from dao (idName(),
+  // stillConnecting(), isWhereByGivenId(),...), may need to be
+  // moved to a more appropriate file (utils.js?)
+  var idPropertyName = dao.idName(Model);
+
+  where[idPropertyName] = id;
+  if (cb === undefined) {
+    if (typeof options === 'function') {
+      // patchById(id, data, cb)
+      cb = options;
+      options = {};
+    }
+  }
+
+  data = data || {};
+  cb = cb || utils.createPromiseCallback();
+  options = options || {};
+
+  assert((typeof data === 'object') && (data !== null),
+    'The data argument must be an object');
+  assert(typeof options === 'object', 'The options argument must be an object');
+  assert(typeof cb === 'function', 'The cb argument must be a function');
+
+  var connector = Model.getDataSource().connector;
+  assert(typeof connector.update === 'function',
```

Comment:
Since we call `connector.updateAttributes` instead of `connector.update`, you should change this check too.